src/wallet/wallet.cpp

```diff
@@ -1425,6 +1425,12 @@ bool CWallet::IsMine(const CTransaction& tx) const
     for (const CTxOut& txout : tx.vout)
         if (IsMine(txout))
             return true;
+    for (const CTxIn& txin : tx.vin) {
+        const CWalletTx* wtx = GetWalletTx(txin.prevout.hash);
+        if (wtx && wtx->tx->vout.size() > txin.prevout.n && IsMine(wtx->tx->vout[txin.prevout.n])) {
+            return true;
+        }
+    }
```

Comment:
The purpose here is to identify the transactions whose inputs belong to the wallet ?

Reply:
Yes

Reply:
This `IsMine` change will have implications inside `AddToWalletIfInvolvingMe` as well.
We are checking tx ownership using the following conditions: `if (fExisted || IsMine(tx) || IsFromMe(tx))`

So, with this changes, the middle `IsMine` will check what `IsFromMe` currently checks. So.. either here or in a follow-up PR, can remove the `IsFromMe` function entirely.

Other point is that you can simplify the code:
```c++
for (const CTxIn& txin : tx.vin) {
    if (IsMine(txin.prevout)) return true; 
}
```

Reply:
I've removed this change and switched to using `IsFromMe` in addition to `IsMine`. Generally we should avoid changing `IsMine` semantics when possible.